src/VkBootstrap.cpp

```diff
@@ -1467,70 +1409,67 @@ detail::Result<Device> DeviceBuilder::build() const {
+
+	if(!user_defined_phys_dev_features_2) {
+		if (physical_device.instance_version >= VK_MAKE_VERSION(1, 1, 0)) {
+			local_features2.features = physical_device.features;
+            final_pnext_chain.push_back(reinterpret_cast<VkBaseOutStructure*>(&local_features2));
+			has_phys_dev_features_2 = true;
+			for (auto& features_node : physical_device_extension_features_copy) {
+				final_pnext_chain.push_back(reinterpret_cast<VkBaseOutStructure*>(&features_node));
+			}
 		}
 	}
 
-// Add the correct structs to the pNext chain if api is 1.1/1.2 and aren't already present
-// This is to guard against users who were already doing that
-#if defined(VK_API_VERSION_1_1)
-	VkPhysicalDeviceFeatures2 local_features2{};
-	if (physical_device.instance_version >= VK_MAKE_VERSION(1, 1, 0) && !has_phys_dev_features_2) {
-		local_features2.sType = VK_STRUCTURE_TYPE_PHYSICAL_DEVICE_FEATURES_2;
-		local_features2.features = physical_device.features;
-		pNext_chain.push_back(reinterpret_cast<VkBaseOutStructure*>(&local_features2));
-		has_phys_dev_features_2 = true;
+	if(!user_defined_phys_dev_features_2 && !has_phys_dev_features_2) {
+        device_create_info.pEnabledFeatures = &physical_device.features;
 	}
-#if defined(VK_API_VERSION_1_2)
-	VkPhysicalDeviceVulkan11Features local_features_11 = physical_device.features_11;
-	VkPhysicalDeviceVulkan12Features local_features_12 = physical_device.features_12;
-	if (physical_device.instance_version >= VK_MAKE_VERSION(1, 2, 0)) {
-		if (!has_phys_dev_vulkan_features_11) {
-			pNext_chain.push_back(reinterpret_cast<VkBaseOutStructure*>(&local_features_11));
-			has_phys_dev_vulkan_features_11 = true;
-		}
-		if (!has_phys_dev_vulkan_features_12) {
-			pNext_chain.push_back(reinterpret_cast<VkBaseOutStructure*>(&local_features_12));
-			has_phys_dev_vulkan_features_12 = true;
-		}
-	}
-#endif
 #endif
 
-	VkDeviceCreateInfo device_create_info = {};
+    for(auto& pnext : info.pNext_chain) {
+        final_pnext_chain.push_back(pnext);
+    }
+
+    detail::setup_pNext_chain(device_create_info, final_pnext_chain);
+    for(auto& node : final_pnext_chain) {
+        assert(node->sType != VK_STRUCTURE_TYPE_APPLICATION_INFO);
+    }
+
 	device_create_info.sType = VK_STRUCTURE_TYPE_DEVICE_CREATE_INFO;
-	detail::setup_pNext_chain(device_create_info, pNext_chain);
 	device_create_info.flags = info.flags;
 	device_create_info.queueCreateInfoCount = static_cast<uint32_t>(queueCreateInfos.size());
 	device_create_info.pQueueCreateInfos = queueCreateInfos.data();
 	device_create_info.enabledExtensionCount = static_cast<uint32_t>(extensions.size());
 	device_create_info.ppEnabledExtensionNames = extensions.data();
-	// VUID-VkDeviceCreateInfo-pNext-00373 - don't add pEnabledFeatures if the phys_dev_features_2 is present
-	if (!has_phys_dev_features_2) {
-		device_create_info.pEnabledFeatures = &physical_device.features;
-	}
 
+	if(!final_pnext_chain.empty()) {
```

Comment:
I am pretty sure this is redundant, `detail::setup_pNext_chain` connects all the structs together and puts the pointer to them inside the passed in struct (device_create_info)